### DISCUSSION TITLE
New name for data_mile1?

### DISCUSSION THREAD

**Discussion by kellijohnson-NOAA** [NOAA-FIMS/FIMS#702] [Ideas]
We need a new, permanent name for data_mile1, which is the name of the data frame stored within FIMS and can be used to run a model. I do not think that the best path forward is to have data sets for each milestone. For lengths, we just added a column and two types. But, an age-structured model could be fit using data_mile1 or a new data frame with the additional length column, where it would just ignore the additional column and age-length-conversion data type. So, I am proposing that we just rename data_mile1 to something better than just "data". Some name that helps users know it is a FIMS data frame for a FIMS model. "fishy_data" does not seem like it fits and "data" is to commonly used within R. We want something that is unique so if we have to do a search and replace, that is easily accomplished without changing a lot of unwanted things. I am sure there are some other considerations that we should be thinking about too. Please provide comments and ideas for a name. We would like to rename the data set tomorrow but if we rename it to something that we end up not liking we have until the end of the month to do that as well. So, throw your ideas out there, we appreciate them all.

**Comment by iantaylor-NOAA** [comment]
How about "data1"? We could eventually have a table of data sets in the documentation with more detailed descriptions of what they contain (along the lines of https://github.com/nmfs-ost/ss3-test-models/tree/main?tab=readme-ov-file#what-features-do-these-models-have). At any point if the set of "data1"..."dataN" becomes too confusing we could easily rename them all at which point we'll have a better idea of how the datasets differ and will therefore be in a better position to choose short informative names to distinguish them from each other.

**Reply by kellijohnson-NOAA** [reply]
Thanks @iantaylor-NOAA for the suggestion. I like how `data1` is unique enough that a search and replace will just grab it and not things named "data", of which there are a lot in our repository. I was slightly worried that it would be unknown that it comes from the FIMS package but this is an unnecessary worry as almost no other package labels the data sets with the package name included and we can always use FIMS::data1 if we are worried about it.

**Reply by nathanvaughan-NOAA** [reply]
dropping the milestone label and going to data1 and data2 sounds great to me.

**Reply by Bai-Li-NOAA** [reply]
About the idea to include a table of datasets in the documentation with more detailed descriptions, we can update the existing `R/data_mile1.R`. All the details about the dataset can be found on the FIMS {pkgdown} site [here](https://noaa-fims.github.io/FIMS/reference/index.html).

**Reply by iantaylor-NOAA** [reply]
@Bai-Li-NOAA, thank you for the reminder, that's far better than some additional table somewhere. I often forgot how good FIMS has been at adopting current best practices for R packages.

**Comment by kellijohnson-NOAA** [comment]
Thanks for the comments on this Discussion. The data_mile1 object and all associated code has been renamed in b6a77b53.
